### ISSUE TITLE
Allow Notepad++ to stay permanently in the system tray

### ISSUE THREAD

**Issue by jez9999** [notepad-plus-plus/notepad-plus-plus#4075]
Notepad++ currently has the ability to minimize to the system tray.  However, I think it would be good to also add the option of "permanently stay in the system tray".  This would mean that the Notepad++ system tray icon would always remain, and when you click the close button, it closes all tabs, closes the main window, but not the system tray icon.  The way to really close Notepad++ would be to right-click on the system tray icon and click "close tray icon".  This would allow you to quickly close all documents in Notepad++, yet still have the convenient system tray icon to open it up again with a blank tab.

**Comment by kgbm3** [comment]
Yes, ofc. "Minimize to system tray" is pretty useless, however close-to-tray (even if it couldn't start up that way): that would be _most_ excellent!..:)

**Comment by adamzovits** [comment]
How is this different from the "Pin to taskbar" function of Windows? (Application can be effectively closed and still at any time be reopened with a single click)

**Comment by MAPJe71** [comment]
@adamzovits depends on the Windows version being used.

**Comment by kgbm3** [comment]
Ugh., plus it's _very_ different; a [pinned application](https://docs.microsoft.com/en-us/windows/configuration/windows-10-start-layout-options-and-policies) will sit in the Taskbar, while something closed to the Tray will **not** clutter the Taskbar (& that application would, technically, be accessible faster - as it isn't terminated). :)

**Comment by bartosz-jarmuz** [comment]
I don't think having it pinned is always fast enough. There can be a delay on launching a new instance, and if you had any remote files open (network drives etc), the delay can be very long. 

This feature would be great!

**Comment by MrChebur** [comment]
I just face the problems when open multiple files with Notepad++ (Windows 10 instantly freeze) described at:

**Comment by michaelcoxon** [comment]
My use-case here is that I keep a large amount of (often unsaved) documents in it and opening can take some time but it uses nothing while open. having the ability to minmise to tray on close would make the it 'open' again instantly and avoid the annoyance of an accidental closure.

**Comment by HrMitrev** [comment]
Please implement that feature. When using np++ with wine on Linux, the startup is very slow and this will fix that problem.

**Comment by dreamforceinc** [comment]
Want this option too. Very want.

**Comment by jez9999** [comment]
Sigh, 3 years after I submitted this issue and still not implemented.  Wouldn't even be that hard for the author to do and would make accessing Notepad++ sooooo much more convenient.

@donho Could you please consider implementing this?

**Comment by sasumner** [comment]
@jez9999 
Notepad++ is a 100% time-donation project by its limited number of developers (and thus not every whim and desire by users can be implemented), and the product is brought to you at no cost.
So maybe stop with the "Sigh"ing.
Notepad++ accepts code submissions; consider donating YOUR time to implement the feature, and submit a PR for it.

**Comment by kanlukasz** [comment]
For Windows users, [RBTray](https://github.com/benbuck/rbtray) can be some kind of solution

**Comment by ZaytsevArtem** [comment]
Some may say 1000-2000 ms (many slow plugins) is not a deal but it's too much for me, so i needed some decision to keep npp always opened. Tried to use macros but this door was also closed as there's no internal command fo _Minimize to tray_.
I made some crutch via AutoHotKey tiny script, and i leave it here in hope it'll help anyone.
This script runs by pressing _Esc_ (you may change it to any other including _Alt+F4_) button when Notepad++ window is active.
It tries to close all tabs and if succeeds, then minimizes the npp window.
This script requires _Settings_->_General_->_**Exit on close all tabs**_ option to be **unchecked**.
A bit bitter is you'll have to edit the code (see the NOTE!-comment) if your npp _Settings_->_General_->_Localization_ is not _English_.
Also you may check the _Settings_->_MISC._->_Minimize to system tray_ option but it is not required for the script to work.
```
;Esc, when Notepad++ window active, tries to close all tabs and if succeeds, then minimizes the app window.
#IfWinActive, ahk_class Notepad++
;For Alt+F4 the line will be «!F4::».
Esc::
;Send Ctrl+Shift+W shortcut for "Close All" command.
Send ^+W
;10 seconds timeout on wait for user reaction should be enough. On timeout the script terminates and npp window stays active.
loopsallowed := 100
;Then, if all tabs are closed successfully, the main window will remain active. Otherwise there will appear a window asking to save a file before closing.
;Wait for user to make a choice. Or not having to make any. Or until timeout.
Loop {
	Sleep 100
	if (!loopsallowed--)
		return
	WinGetClass awc, A
} Until (awc == "Notepad++")
;If the user didn't choose "Cancel", - then all the tabs should be closed, and therefore the main window title now must be "new 1 - Notepad++".
WinGetTitle awt, A
;NOTE! Will have to change it for other language interface, e.g. "новый 1 - Notepad++" for rus). Otherwise the script will not work.
if (awt == "new 1 - Notepad++")
	;WinMinimize command works not as it should so we post WM_SYSCOMMAND = SC_MINIMIZE message.
	PostMessage, 0x112, 0xF020,,, ahk_class Notepad++
return
#IfWinActive
```

**Comment by jez9999** [comment]
@donho For the record, that patch (from what I'm reading of it) doesn't do what I suggested in this issue and I think the issue should be re-opened.  It doesn't permanently keep Notepad++ in the tray (it only goes to the tray on minimize or close), and it doesn't try to close all tabs when close is clicked.  The idea is you should be able to click "minimize" to hide the main window or "close" to close all tabs, and then hide it.

**Comment by donho** [comment]
@jez9999 
I do understand your request. However, for me it's pointless to keep Notepad++ in system tray, since Notepad++ is fast to be launched. OTOH, the implementation of this feature will make the already-complex code even more complex.
As a result, current implementation (option "Close to system tray") is the best compromise I can make. Hope you understand.

**Comment by pawelzwronek** [comment]
@jez9999 I think, a usecase of closing all tabs might be usefull for some people, but imho most would like not to close all tabs, when closing. How about opening new issue to add an option to Preferences "Close all tabs on close".

**Comment by donho** [comment]
> How about opening new issue to add an option to Preferences "Close all tabs on close/exit".

It exists this feature:

![image](https://github.com/user-attachments/assets/d8a347ca-d57a-43ae-b2a4-31910eb58c77)

**Comment by pawelzwronek** [comment]
Right, I forgot. So the new preference could be named "Close all tabs when hiding to tray".

**Comment by pawelzwronek** [comment]
@donho @jez9999  Or how about I make new PR that trigger closing all tabs when `Remember current session for next lunch` is unchecked and make `Remember current session for next lunch` and `Enable session snapshot and periodic backup` independent. Actually they could be independent even without hiding to tray, in scenario when you want close all tabs on exit but still have periodic backups. Then on crash, session would be of course restored.

**Comment by Joyfulbeekeeper** [comment]
Proper "Stay in system tray" function still desired.
Almost any sofware in 2024 has options like:
1) Stay in system tray on "Close" (for whatever reason your have this for "Minimize" while not having for "Close", which is much more rare, though also good option)
2) Launch on Startup
2.1) Silent Launch on Startup in System Tray (if preferred)

**Comment by pawelzwronek** [comment]
@Joyfulbeekeeper 
1. It's already on board: eeb4753
